### PR TITLE
Split out the API request module into its own and expose functions fo…

### PR DESCRIPTION
**Pull request by jsantell**
…r making requests to the main process service. r=nalexander

### CODE REVIEW

File: app/ui/browser/actions/main-actions.js [hunk 1]
```diff
@@ -137,17 +104,12 @@ export function setUserTypedLocation(pageId: string, payload: Object): Action {
...
-      })
-        .then(async function (response) {
-          if (!response.ok) {
-            return;
-          }
-          const body = await response.json();
-          dispatch(profileDiffs.completions(payload.text, body.results));
-        })
-        .catch(); // Right now, nothing to be done.  In the future, we could retry.
+      userAgent.query({ text: payload.text }).then(res => {
```
Comment (ncalexan) [inline app/ui/browser/actions/main-actions.js]: Why not bake this JSON parsing into the UA library?  We should always get some JSON back, even if it's just `{}`.  (We might not, yet.)

File: app/ui/browser/lib/user-agent.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      page,
+    }
+  });
+}
+
+export function createSession({ ancestor, reason }) {
+  return request('/session/start', {
+    method: 'POST',
+    json: {
+      scope: 0,
```
Comment (ncalexan) [inline app/ui/browser/lib/user-agent.js]: Think I'd rather have this from the caller (and defaulted).

File: app/ui/browser/lib/user-agent.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    json: {
+      session,
+      reason,
+    },
+  });
+}
+
+export function createStar({ url, session, title }) {
+  return request(`/stars/${encodeURIComponent(url)}`, {
+    // @TODO Shouldn't this be a POST, in terms of creating
```
Comment (ncalexan) [inline app/ui/browser/lib/user-agent.js]: I thought `PUT` / `DELETE` made more sense than `POST` / `DELETE`.  Not married to it.

**Comment by ncalexan** [comment]
By dropping the `.catch()`, you will see "errors" in the console that you can't really do anything about.  If you're happy with that, roll on.

File: app/ui/browser/actions/main-actions.js (hunk 1 above)
Comment (jsantell) [inline app/ui/browser/actions/main-actions.js]: Initial thought was that's extra processing if we don't want it, but that is trivial -- will add that

File: app/ui/browser/lib/user-agent.js (hunk 3 above)
Comment (jsantell) [inline app/ui/browser/lib/user-agent.js]: `PUT` should be used for updating and `POST` creating, but not familiar with the backend how this is actually represented

**Comment by jsantell** [comment]
w/r/t `catch()`, I'd rather them be visible so we're atleast aware of them -- we shouldn't really be having dropped requests here for the most part

**Pull request closed without merging**